oort/oort-web/src/main/groovy/com/netflix/spinnaker/oort/controllers/ProjectController.groovy

```diff
@@ -0,0 +1,256 @@
+import org.springframework.web.bind.annotation.RestController
+import rx.Observable
+import rx.Scheduler
+import rx.schedulers.Schedulers
+
+import com.netflix.spinnaker.oort.model.ServerGroup.InstanceCounts as InstanceCounts
+
+@Slf4j
+@RestController
+@RequestMapping("/projects/{project}")
+class ProjectController {
+
+  private final Scheduler queryClusterScheduler
+
+  @Autowired
+  ProjectController(@Value('${threadPool.queryCluster:25}') int threadPoolSize) {
+    this(Schedulers.from(newFixedThreadPool(threadPoolSize)))
+  }
+
+  ProjectController(Scheduler queryClusterScheduler) {
+    this.queryClusterScheduler = queryClusterScheduler
+  }
+
+  @Autowired
+  Front50Service front50Service
+
+  @Autowired
+  MessageSource messageSource
+
+  @Autowired
+  List<ClusterProvider> clusterProviders
+
+  @RequestMapping(method= RequestMethod.GET, value = "/clusters")
+  List<ClusterModel> getClusters(@PathVariable String project) {
+    Map projectConfig = null
+    try {
+      projectConfig = front50Service.getProject(project)
+    } catch (e) {
+      log.error("Unable to fetch project (${project})", e)
+      throw new ProjectNotFoundException(name: project)
+    }
+
+    if (projectConfig.config.clusters.size() == 0) {
+      return []
+    }
+
+    List<String> applicationsToRetrieve = projectConfig.config.applications ?: []
+    Map<String, Set<Cluster>> allClusters = retrieveClusters(applicationsToRetrieve)
+
+    projectConfig.config.clusters.findResults { Map projectCluster ->
+      List<String> applications = projectCluster.applications ?: projectConfig.config.applications
+      def applicationModels = applications.findResults { String application ->
+        def appClusters = allClusters[application]
+        Set<Cluster> clusterMatches = findClustersForProject(appClusters, projectCluster)
+        new ApplicationClusterModel(application, clusterMatches)
+      }
+      new ClusterModel(
+          account: projectCluster.account,
+          stack: projectCluster.stack,
+          detail: projectCluster.detail,
+          applications: applicationModels
+      )
+    }
+  }
+
+  private static HashSet<Cluster> findClustersForProject(Set<Cluster> appClusters, Map projectCluster) {
+    if (!appClusters) {
+      return []
+    }
+    appClusters.findAll { appCluster ->
+      Names clusterNameParts = Names.parseName(appCluster.name)
+      appCluster.accountName == projectCluster.account &&
+          nameMatches("stack", clusterNameParts, projectCluster) &&
+          nameMatches("detail", clusterNameParts, projectCluster)
+    }
+  }
+
+  private Map<String, Set<Cluster>> retrieveClusters(List<String> applications) {
+    Map<String, Set<Cluster>> allClusters = [:]
+    def retrievedClusters = Observable.from(applications)
+        .flatMap { application ->
+      retrieveApplication(application).subscribeOn(queryClusterScheduler)
+    }
+    .observeOn(queryClusterScheduler).toList().toBlocking().single()
```

Comment:
Maybe @robfletcher has a suggestion... I think I tried to use `observeOn` in both places, but that made the operations run serially instead of in parallel. Or maybe I tried to use `subscribeOn` in both places. I don't know.